src/functions/index.ts

```diff
@@ -107,6 +108,40 @@ function createComputeFunctionFromDescription(
   return buildComputeFunctionFromDescription(descr);
 }
 
+function tryFormula(
+  computeValueAndFormat: ComputeFunction<Arg, Matrix<ValueAndFormat> | ValueAndFormat>,
+  functionName: string
+): ComputeFunction<Arg, Matrix<ValueAndFormat> | ValueAndFormat> {
+  return function (
+    this: EvalContext,
+    ...args: ComputeFunctionArg<Arg>[]
+  ): Matrix<ValueAndFormat> | ValueAndFormat {
+    try {
+      const computeFormula = computeValueAndFormat.bind(this);
+      return computeFormula(...args);
+    } catch (e) {
+      return handleError(e, functionName);
+    }
+  };
+}
+
+function handleError(e: Error | any, functionName: string): ValueAndFormat {
+  if (!(e instanceof Error)) {
+    e = new Error(e);
+  }
+  if (e instanceof EvaluationError) {
+    e.message = e.message.replace("[[FUNCTION_NAME]]", functionName);
+    return { value: e };
+  }
+  // TODO check this
+  const error = new EvaluationError(
+    e.message.replace("[[FUNCTION_NAME]]", functionName),
+    e?.errorType,
+    e.logLevel !== undefined ? e.logLevel : CellErrorLevel.error
+  );
```

Comment:
can be simplified. `e?.errorType` and `e.logLevel` are always undefined because we know `e` is not an `EvaluationError`
```ts
const error = new EvaluationError(e.message.replace("[[FUNCTION_NAME]]", functionName));
```